payment/schema.py

```diff
@@ -41,6 +44,15 @@ def resolve_payments(self, info, **kwargs):
         show_history = kwargs.get('show_history', False)
         if not show_history and not kwargs.get('uuid', None):
             filters += filter_validity(**kwargs)
+        # OFS-257: Create dynamic filters for the payment mutation
```

Comment:
l38 will block in case the user have only Portal payment,

move the code above to a fonction get_additionnal_filter(additional_filter), this fun
and PermissionDenied should be raised only if "not info.context.user.has_perms(PaymentConfig.gql_query_payments_perms)" and "no filter returned by signals"

with your approach, unless you can compare 2 querrySet, I don't see how you can tell if one of the signal returned anything.

Reply:
Ok. I will do it. This idea is a good one and then I will modify code to add sth like this.